### PR TITLE
add composite rules for squeeze op

### PR DESCRIPTION
**Pull request by warrentdrew**
### PR types

New features
### PR changes

APIs
### Describe

add composite rules for squeeze op
no testcast for fp16 for squeeze

### CODE REVIEW

File: python/paddle/incubate/autograd/composite_rules.py [hunk 1]
```diff
@@ -332,6 +332,39 @@ def fill_any_like(x, fill_value, dtype, place=None):
     return val
 
 
+@REGISTER_COMPOSITE('squeeze2')
+def squeeze2_composite(x, axis):
+    """define composite rule of squeeze"""
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/composite_rules.py:441]: describe composite logic details

File: python/paddle/incubate/autograd/composite_rules.py [hunk 2]
```diff
@@ -332,6 +332,45 @@ def fill_any_like(x, fill_value, dtype, place=None):
...
+            return canonicalize_dim(rank, indices)
+        return tuple(canonicalize_dim(rank, idx) for idx in indices)
+
+    if len(axis) == 0:
+        dims = range(len(x.shape))
+    else:
+        dim = canonicalize_dims(len(x.shape), axis)
+        dims = set((dim,) if not isinstance(dim, tuple) else dim)
+    new_shape = []
+    for d, s in enumerate(x.shape):
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/composite_rules.py:455]: can the logic be simplified?

File: python/paddle/incubate/autograd/composite_rules.py [hunk 3]
```diff
@@ -332,6 +332,32 @@ def fill_any_like(x, fill_value, dtype, place=None):
...
+def squeeze2_composite(x, axis):
+    """define composite rule of squeeze"""
+    """
+    canonicalize dim within range 0 to rank and
+    determine new shape after squeeze op
+    if axis not specified, remove all dims equal to 1
+    otherwise, remove dims equal to 1 in axis
+    """
+
+    if len(axis) == 0:
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/composite_rules.py:450]: if axis is int type?

File: python/paddle/incubate/autograd/composite_rules.py [hunk 4]
```diff
@@ -420,6 +420,31 @@ def fill_any_like(x, fill_value, dtype, place=None):
...
+    """define composite rule of squeeze"""
+    """
+    canonicalize dim within range 0 to rank and
+    determine new shape after squeeze op
+    if axis not specified, remove all dims equal to 1
+    otherwise, remove dims equal to 1 in axis
+    """
+    rank = len(x.shape)
+    if isinstance(axis, int):
+        dims = set((axis + rank) % rank)
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/composite_rules.py]: (axis + rank) % rank  equals to axis%rank ?

**Review by cyber-pioneer — APPROVED** [review]

**Review by Charles-hit — APPROVED** [review]

**Pull request merged**